client/state/plugins/installed/actions.js

```diff
@@ -532,3 +535,32 @@ export function fetchSitePlugins( siteId ) {
 export function fetchPlugins( siteIds ) {
 	return ( dispatch ) => siteIds.map( ( siteId ) => dispatch( fetchSitePlugins( siteId ) ) );
 }
+
+export function fetchAllPlugins() {
+	return ( dispatch ) => {
+		dispatch( { type: ALL_PLUGINS_REQUEST } );
+
+		const receivePluginsDispatchSuccess = ( { sites } ) => {
+			dispatch( { type: ALL_PLUGINS_REQUEST_SUCCESS } );
+
+			Object.entries( sites ).map( ( [ siteId, plugins = [] ] ) => {
+				dispatch( receiveSitePlugins( siteId, plugins ) );
+
+				plugins.map( ( plugin ) => {
+					if ( plugin.update && plugin.autoupdate ) {
+						updatePlugin( siteId, plugin )( dispatch );
+					}
```

Comment:
A question out of curiosity. Does this mean that if a plugin needs an update and has auto-update enabled, the frontend will dispatch a request to update the plugin?

Reply:
That's right. I'm not sure about why it works this way, though. @tyxla, do you have any thoughts to share here? I'm curious what would happen if the user never visits the plugin management UI, do we have any fallback mechanism for running auto-updates?

Reply:
We could remove that if we want to. We're currently triggering an async job to auto-update plugins after each sync action, look for `maybe_autoupdate` in the WP.com codebase and you'll find it.

I believe this is here because the async job is not guaranteed to run anytime soon, though. Whether this way to force autoupdate is a good UX, that's arguable.